packages/sources/coinmarketcap-test/src/endpoint/crypto.ts

```diff
@@ -1,25 +1,270 @@
+        requests.push({
+          params: list,
+          request: {
+            baseURL: settings.API_ENDPOINT,
+            url: '/cryptocurrency/quotes/latest',
+            headers: {
+              'X-CMC_PRO_API_KEY': settings.API_KEY,
+            },
+            params: {
+              [idType]: [...new Set(list.map((p) => p.cid || p.slug || p.base))].join(','),
+              convert: [...new Set(list.map((p) => p.quote))].join(','),
+            },
+          },
+        })
+      }
+    }
+
+    return requests
   },
   parseResponse: (params, res) => {
-    return constructEntry(params, res.data, 'price')
+    logger.debug(`CMC api call cost: ${res.data.cost}`)
+
+    // The params are already grouped by ID type, so we only have to look at the first one
+    const idType = params[0].cid ? 'id' : params[0].slug ? 'slug' : 'symbol'
+
+    // Build a mapping to ids
+    const idsMap: Record<string, string> = {}
+    for (const data of Object.values(res.data.data)) {
+      idsMap[data[idType]] = data.id.toString()
+    }
+
+    // Use the mapping to generate the responses
+    return params.map((p) => {
+      const id = idsMap[p.cid || p.slug || p.base]
+      if (!id) {
+        return {
+          params: p,
+          response: {
+            statusCode: 502,
+            errorMessage: `An ID was not found in the idsMap for parameter ${JSON.stringify(p)}`,
+          },
+        }
+      }
+
+      const data = res.data.data[id]
+      if (!data) {
+        return {
+          params: p,
+          response: {
+            statusCode: 502,
+            errorMessage: `ID type "${idType}" was not found in response for request: ${JSON.stringify(
+              p,
+            )}`,
+          },
+        }
+      }
+
+      const dataForQuote = data.quote[p.quote]
+      if (!dataForQuote) {
+        return {
+          params: p,
+          response: {
+            statusCode: 502,
+            errorMessage: `Data for quote "${
+              p.quote
+            }" was not found in response for request: ${JSON.stringify(p)}`,
+          },
+        }
+      }
+
+      const valueRequested = dataForQuote[p.resultPath]
+      if (valueRequested == null) {
+        return {
+          params: p,
+          response: {
+            statusCode: 502,
+            errorMessage: `Value for "${
+              p.resultPath
+            }" was not found in the quote request: ${JSON.stringify(p)}`,
+          },
+        }
+      }
+
+      // We're adding multiple results because the same provider endpoint provides values for several adapter endpoints
+      // Price
+      return {
+        params: p,
+        response: {
+          result: dataForQuote[p.resultPath],
+          data: {
+            result: dataForQuote[p.resultPath],
+          },
+        },
+      }
+    })
   },
 })
 
 export const endpoint = new CryptoPriceEndpoint<CryptoEndpointTypes>({
-  name: 'crypto',
-  aliases: ['price'],
+  name: 'price',
```

Comment:
I know it's covered by the CryptoPriceEndpoint but looks like the name and alias got switched around here